crates/ruff/src/printer.rs

```diff
@@ -264,6 +269,9 @@ impl Printer {
                 self.write_summary_text(writer, diagnostics)?;
             }
             SerializationFormat::Grouped => {
+                if self.flags.intersects(Flags::SHOW_SOURCE) {
+                    warn_user!("`--show-source` with `--output-format=grouped` is deprecated. Source display will not be available for `--output-format=grouped` in future releases.");
+                }
                 GroupedEmitter::default()
                     .with_show_source(self.flags.intersects(Flags::SHOW_SOURCE))
```

Comment:
I'm wondering if we should get rid of this flag, and instead match against `SerializationFormat::Full`?